modules/SofaGraphComponent/RequiredPlugin.cpp

```diff
@@ -42,48 +42,93 @@ int RequiredPluginClass = core::RegisterObject("Load the required plugins")
+    helper::vector<std::string> suffixVec;
+    if (!sMap.empty())
+    {
+        std::string skey = (defaultSuffix.empty() ? std::string("!") : defaultSuffix);
+        for (std::size_t i = 0; i < sMap.size(); ++i)
+        {
+            if (sMap[i][0] == skey)
+            {
+                suffixVec.push_back(sMap[i][1] == std::string("!") ? std::string(""):sMap[i][1]);
+            }
+        }
+    }
+    if (suffixVec.empty())
+        suffixVec.push_back(defaultSuffix);
+    const helper::vector<std::string>& nameVec = d_pluginName.getValue();
+    helper::vector<std::string> nameVecCopy=nameVec;
+    if(nameVec.empty()) nameVecCopy.push_back(this->getName());
+    helper::vector< std::string > loaded;
+    helper::vector< std::string > failed;
+    std::ostringstream errmsg;
+    for (std::size_t nameIndex = 0; nameIndex < nameVecCopy.size(); ++nameIndex)
     {
-        if (!PluginManager::getInstance().pluginIsLoaded(path))
+        const std::string& name = nameVecCopy[nameIndex];
+        //sout << "Loading " << name << sendl;
+        bool nameLoaded = false;
+        for (std::size_t suffixIndex = 0; suffixIndex < suffixVec.size(); ++suffixIndex)
         {
-            if (PluginManager::getInstance().loadPlugin(path))
+            const std::string& suffix = suffixVec[suffixIndex];
+            std::string pluginPath = pluginManager->findPlugin(name, suffix, false);
+            bool result = !pluginPath.empty();
+            if (result && !pluginManager->pluginIsLoaded(pluginPath))
+            {
+                result = pluginManager->loadPlugin(pluginPath, suffix, false, &errmsg);
+            }
+            if (result)
             {
-                const std::string guiPath = pluginManager.findPlugin(pluginName + "_" + PluginManager::s_gui_postfix);
-                if (guiPath != "")
-                {
-                    PluginManager::getInstance().loadPlugin(guiPath);
-                }
+                sout << "Loaded " << pluginPath << sendl;
+                loaded.push_back(pluginPath);
+                nameLoaded = true;
+                if (d_stopAfterFirstSuffixFound.getValue()) break;
             }
         }
+        if (!nameLoaded)
+        {
+            failed.push_back(name);
+        }
+        else
+        {
+            if (d_stopAfterFirstNameFound.getValue()) break;
+        }
     }
-    else
+    if (!failed.empty())
     {
-        msg_error("RequiredPlugin") << "Plugin not found: \"" + pluginName + "\"";
+        if ((d_requireAll.getValue() || (d_requireOne.getValue() && loaded.empty())))
+        {
+            serr << errmsg.str();
+            serr << "Required plugin"<<(failed.size()>1?"s":"")<<" failed to load: " << failed << sendl;
+        }
+        else
+        {
+            sout << errmsg.str();
+            sout << "Optional/alternate plugin"<<(failed.size()>1?"s":"")<<" failed to load: " << failed << sendl;
+        }
```

Comment:
same as before, about the message API

Reply:
This one comes from our local version. We do not use the message API at InSimo